### PR TITLE
Better test suite

### PR DESCRIPTION
**Pull request by assembler-0**
## Summary by CodeRabbit

* **New Features**
  * More reliable, asynchronous capture of process stdout/stderr.
  * Improved cross-platform command discovery honoring PATH/PATHEXT and extensions.

* **Bug Fixes**
  * Eliminates deadlocks during process output reading; execution time reported after I/O completes.
  * Correct executable checks and robust PATH traversal.
  * Predictable environment merging with overrides (case-insensitive on Windows).

* **Tests**
  * Added Linux-only tests for large output, relative exec, permissions, and env merging.

* **Chores**
  * CI simplified to a single Ubuntu build-and-test workflow.

### CODE REVIEW

File: Evaluate.cpp [hunk 1]
```diff
@@ -191,7 +193,7 @@ void TestEdgeCases(TestRunner& runner) {
     std::cout << "\n=== Edge Case Tests ===" << std::endl;
     
     // Empty command
-    std::vector<std::string> empty_args;
+    constexpr std::vector<std::string> empty_args;
     runner.Assert(!ExecutionValidator::CanExecuteCommand(empty_args), "Empty args rejected");
```
[the remark this answers is not in the document]
Comment (assembler-0) [reply Evaluate.cpp:197]: but we are C++20 only so is it still fine?
